connectors/translator-swagger/src/main/java/org/teiid/translator/swagger/SwaggerProcedureExecution.java

```diff
@@ -0,0 +1,546 @@
+            }
+            
+            Clob headers = (Clob)arguments.get(arguments.size() - 1).getArgumentValue().getValue();
+            if(null != headers) {
+                parseHeader(httpHeaders, headers);
+            }
+            
+            applyAuthParams(procedure, httpHeaders);
+            
+            headersValidation(method, httpHeaders, procedure);
+            
+            DataSource ds = null;
+            if (payload instanceof String) {
+                ds = new InputStreamFactory.ClobInputStreamFactory(new ClobImpl((String)payload));
+            } else if (payload instanceof SQLXML) {
+                ds = new InputStreamFactory.SQLXMLInputStreamFactory((SQLXML)payload);
+            } else if (payload instanceof Clob) {
+                ds = new InputStreamFactory.ClobInputStreamFactory((Clob)payload);
+            } else if (payload instanceof Blob) {
+                ds = new InputStreamFactory.BlobInputStreamFactory((Blob)payload);
+            }
+
+            this.returnValue = dispatch.invoke(ds);
+            
+            responseContextValidation(httpHeaders.get("Accept"), dispatch.getResponseContext()); //$NON-NLS-1$  
+            
+            handleReturnValue(procedure, returnValue);
+        } catch (Exception e) {
+            throw new TranslatorException(SwaggerPlugin.Event.TEIID28010, SwaggerPlugin.Util.gs(SwaggerPlugin.Event.TEIID28010, e));
+        }
+ 
+    }
+
+
+    private void handleReturnValue(Procedure procedure, DataSource returnValue) throws TranslatorException {
+        
+        try {
+            if (procedure.getResultSet() != null) {
+                ResultsType resultsType = getReturnType(procedure);
+                this.jsonResponse = new JsonResponse(this.returnValue.getInputStream(), resultsType);
+            } else if (getReturnParameter(procedure) != null) {
+                if(this.returnValue.getContentType().equals("application/json")){ //$NON-NLS-1$
+                    result = parser.toJsonString(this.returnValue.getInputStream());
+                } else if(this.returnValue.getContentType().equals("application/xml")) { //$NON-NLS-1$
+                    result = parser.toXmlString(this.returnValue.getInputStream());
+                }
+                if(result == null) {
+                    result = new BlobType(new StreamingBlob(this.returnValue.getInputStream()));
+                }
+            }
+        } catch (IOException e) {
+            throw new TranslatorException(e);
+        }
+        
+    }
+
+    private Object getReturnParameter(Procedure procedure) {
+        for (ProcedureParameter pp : procedure.getParameters()){
+            if (pp.getType() == ProcedureParameter.Type.ReturnValue) {
+                return pp;
+            }
+        } 
+        return null;
+    }
+
+    private void applyAuthParams(Procedure procedure, Map<String, List<String>> httpHeaders) throws TranslatorException {
```

Comment:
We can not have any Auth specific code in the translator. It needs be at resource-adapter level. You can have metadata on the translator that it needs to use certain Authentication but not enforce at this layer. So remove all auth specific code.